Sources/Basic/TerminalController.swift

```diff
@@ -0,0 +1,136 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import libc
+import func POSIX.getenv
+
+/// Tries to get the terminal width first using COLUMNS env variable and
+/// if that fails ioctl method testing on stdout stream.
+///
+/// - Returns: Current width of terminal if it was determinable.
+public func terminalWidth() -> Int? {
+    // Try to get from enviornment.
+    if let columns = POSIX.getenv("COLUMNS"), let width = Int(columns) {
+        return width
+    }
+
+    // Try determining using ioctl.
+    var ws = winsize()
+    if ioctl(1, UInt(TIOCGWINSZ), &ws) == 0 {
+        return Int(ws.ws_col)
+    }
+    return nil
+}
+
+/// Terminal color choices.
+public enum TermColor {
```

Comment:
Yeah moving inside class makes more sense, initially I think I couldn't because it was generic but now should be possible